src/x509.c

```diff
@@ -7517,6 +7517,464 @@ WOLFSSL_X509_CRL* wolfSSL_d2i_X509_CRL(WOLFSSL_X509_CRL** crl,
+{
+    WOLFSSL_ENTER("wolfSSL_X509_CRL_get_signature");
+    if (crl == NULL || buf == NULL || bufSz == NULL)
+        return WOLFSSL_FATAL_ERROR;
+
+    if (buf != NULL)
+        XMEMCPY(buf, crl->crlList->signature, *bufSz);
+
+    return WOLFSSL_SUCCESS;
+}
+
+int wolfSSL_X509_REVOKED_get_serial_number(RevokedCert* rev,
+                                            byte* in, int* inOutSz)
+{
+    WOLFSSL_ENTER("wolfSSL_X509_CRL_get_serial_number");
+    if (rev == NULL || inOutSz == NULL) {
+        WOLFSSL_MSG("Null argument passed in");
+        return BAD_FUNC_ARG;
+    }
+
+    if (in != NULL) {
+        if (*inOutSz < rev->serialSz) {
+            WOLFSSL_MSG("Serial buffer too small");
+            return BUFFER_E;
+        }
+        XMEMCPY(in, rev->serialNumber, rev->serialSz);
+    }
+    *inOutSz = rev->serialSz;
+
+    return WOLFSSL_SUCCESS;
+}
+
+
+int wolfSSL_X509_CRL_print(WOLFSSL_BIO* bio, WOLFSSL_X509_CRL* crl)
+{
+    if (bio == NULL || crl == NULL) {
+        return WOLFSSL_FAILURE;
+    }
+
+    if (wolfSSL_BIO_write(bio, "Certificate Revocation List (CRL):\n",
+                  (int)XSTRLEN("Certificate Revocation List (CRL):\n")) <= 0) {
+            return WOLFSSL_FAILURE;
+    }
+
+    /* print version */
+    {
+        int version;
+        char tmp[20];
+
+        if ((version = wolfSSL_X509_CRL_version(crl)) < 0) {
+            WOLFSSL_MSG("Error getting CRL version");
+            return WOLFSSL_FAILURE;
+        }
+        if (wolfSSL_BIO_write(bio, "        Version:",
+                      (int)XSTRLEN("        Version:")) <= 0) {
+            return WOLFSSL_FAILURE;
+        }
+        if (XSNPRINTF(tmp, sizeof(tmp), " %d (0x%x)\n",
+                      version, (byte)version-1)
+            >= (int)sizeof(tmp))
+        {
+            return WOLFSSL_FAILURE;
+        }
+        if (wolfSSL_BIO_write(bio, tmp, (int)XSTRLEN(tmp)) <= 0) {
+            return WOLFSSL_FAILURE;
+        }
+    }
+       
+    /* print signature algo */
+    {
+        int oid;
+        const char* sig;
+
+        if ((oid = wolfSSL_X509_CRL_get_signature_type(crl)) <= 0) {
+            WOLFSSL_MSG("Error getting x509 signature type");
+            return WOLFSSL_FAILURE;
+        }
+        if (wolfSSL_BIO_write(bio, "        Signature Algorithm: ",
+                      (int)XSTRLEN("        Signature Algorithm: ")) <= 0) {
+            return WOLFSSL_FAILURE;
+        }
+
+        sig = GetSigName(oid);
+        if (wolfSSL_BIO_write(bio, sig, (int)XSTRLEN(sig)) <= 0) {
+            return WOLFSSL_FAILURE;
+        }
+        if (wolfSSL_BIO_write(bio, "\n", (int)XSTRLEN("\n")) <= 0) {
+            return WOLFSSL_FAILURE;
+        }
+    }
+    
+    {
+        WOLFSSL_X509_NAME* name = wolfSSL_X509_CRL_get_issuer_name(crl);
+        char* issuer = name->name;
+
+        if (wolfSSL_BIO_write(bio, "        Issuer: ",
+                      (int)XSTRLEN("        Issuer: ")) <= 0) {
+#ifdef WOLFSSL_SMALL_STACK
```

Comment:
Alignment issue here. Please confirm this small stack logic is valid. Make sure and test with --enable-smallstack.